worker/internal/usecase/gateway/file.go

```diff
@@ -0,0 +1,22 @@
+package gateway
+
+import (
+	"context"
+	"errors"
+	"io"
+	"net/url"
+)
+
+var (
+	ErrInvalidFile        error = errors.New("invalid file")
+	ErrNotFound           error = errors.New("not found")
+	ErrFailedToUploadFile error = errors.New("failed to upload file")
+	ErrFileTooLarge       error = errors.New("file too large")
+	ErrFailedToRemoveFile error = errors.New("failed to remove file")
+)
+
+type File interface {
+	// ReadAsset(ctx context.Context, name string) (io.ReadCloser, error)
+	RandomReadAssetByURL(ctx context.Context, url *url.URL) (io.ReaderAt, int64, error)
+	UploadAssetFunc(ctx context.Context, name string) (io.WriteCloser, error)
+}
```

Comment:
```suggestion

type ReadAtCloser interface {
	io.ReaderAt
	io.Closer
}

type File interface {
    Read(ctx context.Context, path string) (io.ReadAtCloser, int64, error)
	Upload(ctx context.Context, name string) (io.WriteCloser, error)
}
```

- There is no way to disconnect, causing connection leaks. io.ReaderAt with io.Closer would be needed.
- File does not have any knowledge of Asset. Therefore, it is unnatural for Asset to be used as a method name in File.
- Simpler method names. Imagene usage: `file.Upload()`. This is enough to know what is being done.
- `Read` should accept a path string instead of a URL. File has no knowledge of where it exists in the first place. It does not know whether it is in the GCS or a local file. It is only interested in the its own file system that exists somewhere. And the only way to express the location of a file in the file system is by path. If the File accept URLs, it have to deal with extra information about where acually it is.

Reply:
>Read should accept a path string instead of a URL. File has no knowledge of where it exists in the first place. It does not know whether it is in the GCS or a local file. It is only interested in the its own file system that exists somewhere. And the only way to express the location of a file in the file system is by path. If the File accept URLs, it have to deal with extra information about where acually it is.

hmm, I see, I agree with that since the hostname of them can also be changed, and using the path is more flexible for the change.  

Question: Why Re:Earth uses `url.URL` [these part](https://github.com/reearth/reearth/blob/c7e42351b0af901dd6060f3555c869e7103929a0/server/internal/usecase/gateway/file.go#L22)? Are these expected to refactored in the future?